### DISCUSSION TITLE
Quick Look not working on iOS when model viewer is used inside iframe

### DISCUSSION THREAD

**Discussion by kbs1** [google/model-viewer#4139] [Q&A] [answered]
Quick look opens on iPhone, but model never loads. All works properly when iframe is not used. Iframe code:
```
<iframe src="...page with model-viewer 3.x..." style="border:0 none;width:150px;height:150px" allowvr="yes" allow="xr-spatial-tracking; accelerometer; magnetometer; gyroscope; camera; autoplay" allowfullscreen="yes" allowtransparency="true"></iframe>
```
Is this expected to work, or are the any workarounds?

**Comment by elalish** [comment, marked as answer]
I think that should work, but iOS Safari has some strange bugs. Do you see anything in the JS console?

**Reply by kbs1** [reply]
Thank you for the clarification. I've connected the debugger and iframe was not the issue, it's the GLB model itself:
```
[Warning] THREE.USDZExporter: Unsupported material type (USDZ only supports MeshStandardMaterial)
```
We have USDZ files so we will pass them as ios-src to ensure Quick look can always be used.
